src/test/java/com/stripe/functional/CardTest.java

```diff
@@ -1,20 +1,150 @@
+    stubRequest(
+        APIResource.RequestMethod.POST,
+        String.format("/v1/customers/%s/sources", customer.getId()),
+        params,
+        Card.class,
+        getResourceAsString("/api_fixtures/card.json")
+    );
+
+    final Card card = (Card) customer.getSources().create(params);
+
+    assertNotNull(card);
+    verifyRequest(
+        APIResource.RequestMethod.POST,
+        String.format("/v1/customers/%s/sources", customer.getId()),
+        params
+    );
+  }
+
+  @Test
+  public void testRetrieve() throws IOException, StripeException {
+    final Customer customer = Customer.retrieve(CUSTOMER_ID);
+
+    // stripe-mock returns a BankAccount instance instead of a Card
+    stubRequest(
+        APIResource.RequestMethod.GET,
+        String.format("/v1/customers/%s/sources/%s", customer.getId(), CARD_ID),
+        null,
+        Card.class,
+        getResourceAsString("/api_fixtures/card.json")
+    );
+
+    final Card card = (Card) customer.getSources().retrieve(CARD_ID);
+
+    assertNotNull(card);
+    verifyRequest(
+        APIResource.RequestMethod.GET,
+        String.format("/v1/customers/%s/sources/%s", customer.getId(), CARD_ID)
+    );
+  }
+
   @Test
-  public void testCardMetadata() throws StripeException {
-    Customer customer = Customer.create(defaultCustomerParams);
-    Map<String, Object> creationParams = new HashMap<String, Object>();
-    creationParams.put("source", "tok_visa");
-    testMetadata(customer.createCard(creationParams));
+  public void testUpdate() throws IOException, StripeException {
+    final Customer customer = Customer.retrieve(CUSTOMER_ID);
+    final Card card = getCardFixture(customer);
+
+    Map<String, Object> metadata = new HashMap<String, Object>();
+    metadata.put("key", "value");
+    Map<String, Object> params = new HashMap<String, Object>();
+    params.put("metadata", metadata);
+
+    final Card updatedCard = card.update(params);
+
+    assertNotNull(updatedCard);
+    verifyRequest(
+        APIResource.RequestMethod.POST,
+        String.format("/v1/customers/%s/sources/%s", customer.getId(), card.getId()),
+        params
+    );
+  }
+
+  @Test
+  public void testList() throws IOException, StripeException {
+    final Customer customer = Customer.retrieve(CUSTOMER_ID);
+
+    Map<String, Object> params = new HashMap<String, Object>();
+    params.put("object", "card");
+
+    // stripe-mock doesn't handle this, so we stub the request
+    Card stubbedCard = getCardFixture(customer);
+    ExternalAccountCollection stubbedCollection = new ExternalAccountCollection();
+    List<ExternalAccount> stubbedData = new LinkedList<ExternalAccount>();
+    stubbedData.add(stubbedCard);
+    stubbedCollection.setData(stubbedData);
+    stubRequest(
+        APIResource.RequestMethod.GET,
+        String.format("/v1/customers/%s/sources", customer.getId()),
+        params,
+        ExternalAccountCollection.class,
+        stubbedCollection.toJson()
+    );
```

Comment:
same as with the bank account, why not a real fixture?

Reply:
same as above